File: web_new/src/types/card.ts
// Card data types based on existing API structure

export interface ReplyCard {
  cid: number;
  time: number;
  difficulty: number;
  stability: number;
  decay: number;
  front: string;
  back: string;
  paused?: true;
}

export interface ReplyGroup {
  group: string[];
}

export type ApiResponseItem = ReplyGroup | ReplyCard;

export interface CardData extends Omit<ReplyCard, 'cid'> {
  groupPath: string[];
  retentionRate: number;
  uniqueId: string;
}

export interface CardGroup {
  name: string;
  path: string[];
  cards: CardData[];
  subgroups: Map<string, CardGroup>;
  totalCards: number;
  averageRetention: number;
}

// Group list item for the hierarchical group display
export interface GroupListItem {
  name: string;
  path: string[];
  totalCards: number;
  averageRetention: number;
  subgroups: GroupListItem[];
  isExpanded: boolean;
}

File: ankirus/models/card.py
"""卡片数据模型"""

from dataclasses import dataclass
from typing import Literal, NotRequired, TypedDict


@dataclass
class Card:
    """卡片数据类"""

    cid: int
    group: str
    front: str
    back: str
    time: int  # Unix Timestamp
    difficulty: float
    stability: float
    decay: float
    paused: bool
    due: int  # Unix Timestamp


class ReplyGroup(TypedDict):
    """API响应分组格式"""

    group: list[str]


class ReplyCard(TypedDict):
    """API响应卡片格式"""

    time: int
    difficulty: float
    stability: float
    decay: float
    front: str
    back: str
    paused: NotRequired[Literal[True]]
